src/parser.c

```diff
@@ -2001,84 +2009,141 @@ fgetline (FILE *fp) {
   return NULL;
 }
 
-/* Iterate over the log and read line by line (use GNU get_line to parse the
- * whole line).
+/* Parse chunk of lines to logitems */
+void *
+read_lines_thread (void *arg) {
+  GJob *job = (GJob *) arg;
+  for (int i=0; i<job->p; i++) {
+    job->logitems[i] = read_line (job->glog, job->lines[i], &job->test, &job->cnt, job->dry_run);
+
+#ifdef WITH_GETLINE
+    free (job->lines[i]);
+#endif
+  }
+  pthread_exit((void *) 0);
```

Comment:
```suggestion
  return (void *) 0;
```
This should assist in preventing the omission of cleanup/destructors flagged by Valgrind.

Reply:
Oh, I forget to run Valgrind (use ASAN instead).
Thanks! Will rebase before merge.

Reply:
No worries. I strongly suggest running it through helgrind, e.g,

    valgrind --tool=helgrind ./goaccess access.log --log-format=VCOMBINED -o report.html -j 3

It appears there are some data races that might require addressing with locks.

Reply:
Found some data races with helgrind and fixed those!